vulkaninfo/CMakeLists.txt

```diff
@@ -1,5 +1,89 @@
 # CMakeLists.txt file for building Vulkaninfo
 
+set(VULKANINFO_INCLUDE_DIRS
+    ${CMAKE_CURRENT_BINARY_DIR}
+    ${CMAKE_CURRENT_BINARY_DIR/..}
+   )
+
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    include(FindPkgConfig)
+    option(BUILD_WSI_XCB_SUPPORT "Build XCB WSI support" ON)
+    option(BUILD_WSI_XLIB_SUPPORT "Build Xlib WSI support" ON)
+    option(BUILD_WSI_WAYLAND_SUPPORT "Build Wayland WSI support" ON)
+    option(BUILD_WSI_MIR_SUPPORT "Build Mir WSI support" ON)
+    set(VULKANINFO_WSI_SELECTION "XCB" CACHE STRING "Select WSI target for vulkaninfo (XCB, XLIB, WAYLAND, MIR, DISPLAY)")
+
+    if (BUILD_WSI_XCB_SUPPORT)
+        find_package(XCB REQUIRED)
+    endif()
+
+    if (BUILD_WSI_XLIB_SUPPORT)
+        find_package(X11 REQUIRED)
+    endif()
+    
+    if (BUILD_WSI_WAYLAND_SUPPORT)
+        find_package(Wayland REQUIRED)
+
+        include_directories(${WAYLAND_CLIENT_INCLUDE_DIR})
```

Comment:
If wayland needs includes to even build (which is what this looks like), would it be better to use this snippet from below:
````
+        set(VULKANINFO_INCLUDE_DIRS
+            ${WAYLAND_CLIENT_INCLUDE_DIRS}
+            ${VULKANINFO_INCLUDE_DIRS}
+        )
````
Then you can skip doing it below because it will already be done.